src/Main.enso

```diff
@@ -1,107 +1,190 @@
 
-    # Method calls – `.` calls method `sort` of `operator1`
-    # We can pass additional arguments
-    operator1.sort Ascending
+    # Assignments
+    # `=` for assignments, storing the result of the right-hand side under
+    # the name on the left.
+    x = 1
+    y = 10
 
-    # Operators have different precedence depending on spacing
-    # No-spaces binds stronger than a spaced version
-    # foo bar . baz ==> (foo bar).baz
-    # foo bar.baz   ==> foo (bar.baz)
-    operator1 = 0.up_to 100 . to_vector . map .noise
+    # Enso has single assignments: you cannot modify existing variable.
+    # x = 2 now will result in a compile error.
 
-    # Operator sections!
+    # Methods can be called with `.`.
+    # Methods can be passed additional arguments.
+    # `up_to` creates a `Range` from 1 up to 10.
+    range = x.up_to y
+
+    # Operator sections:
     # (+ 5) => a function that takes x and returns x+5
-    # (5 +) => ...
+    # (5 +) => a function that takes x and returns 5+x
     # [1, 2, 3] . map (+ 5)
     # [1, 2, 3] . map +5
     # .noise => a function that takes x and return x.noise
-    operator1.sort
 
-    # Lambdas
+    # Operators, including `.`, have different precedence depending on spacing
+    # No-spaces binds stronger than a spaced version
+    # foo bar . baz ==> (foo bar).baz
+    # foo bar.baz   ==> foo (bar.baz)
+    result_2 = 2+2 * 2 # (2+2)*2 == 8
+    result_3 = 2 + 2 * 2 # 6, conventional aritmetic operator
+    mapped_list = range . to_vector . map .noise
+    # equivalent to:
+    # mapped_list = (range.to_vector).map .noise
+
+    # Lambdas / anonymous functions
     # Created with `->`
-    # Arguments go on the left, result goes on the right
+    # Parameters on the left, results on the right:
     [1, 2, 3] . map (x -> x * 5)
-    [1, 2, 3] . map z-> y-> x->
-        y_1 = x * 5
-        z_1 = y_1 + 3
-        z_1
+    # Multiple parameters are achieved by multiple `->`:
     [1, 2, 3] . reduce (x -> y -> x + y)
 
     # Lambda shorthands
-    my_lambda = (here.my_method _ 5) # equivalent to (x-> here.my_method x 5)
-    my_lambda_2 = (here.my_method _ 6 _ 5) # equivalent to (y-> x-> here.my_method y 6 x 5)
+    my_lambda = (here.sum _ 5) # equivalent to (x-> here.sum x 5)
 
     # Calling a method
-    this.my_method 1 2 # `this` refers to whatever the current method is called on
-    here.my_method 1 2 # `here` ALWAYS refers to the current module
+    this.sum 1 2 # `this` refers to whatever the current method is called on
+    here.sum 1 2 # `here` ALWAYS refers to the current module
 
-    # Calling a method with default arguments, by name
-    my_circ_1 = here.draw_circle x_position=37 radius=7 # equivalent to here.draw_circle 7 'red' 37 0
-    my_circ_2 = here.draw_circle 36 'blue' # equivalent to here.draw_circle 36 'blue' 0 0
-    my_circ_factory = here.draw_circle 36 'blue' ... # equivalent to (x=0)-> (y=0)-> here.draw_circle 36 'blue' x y
+    # Methods with default parameters can be called
```

Comment:
@kustosz I think something got cut off here, but actually this gets to a question I had about one of your comments last week, when you were talking about the sort API. 

`Methods with default parameters can be called with named parameters, or without` is this accurate? And if so, like our example here suggests, why is this the case? https://github.com/enso-org/learnxinyminutes/pull/2#discussion_r569285025

Reply:
Ah I see the confusion. Yes, they can be called with or without named parameters, always. Regarding my comment "this API is designed for use with named arguments" that's purely a matter of taste. In this case, the `sort` method takes 3 defaulted arguments (the ordering, a comparator function, and a preprocessor function) and you want to provide any number of these. It would be horrible to have to remember this order of arguments, while their names are intuitive.
It's just about the way the code looks, no technical limitation. Examples:
```
[my_foo, my_bar, my_baz].sort
[my_foo, my_bar, my_baz].sort order=Order.Ascending
[my_foo, my_bar, my_baz].sort by=(.my_property) order=Order.Ascending
[my_foo, my_bar, my_baz].sort order=Order.Ascending  by=(.my_property)
[my_foo, my_bar, my_baz].sort by=(.my_property)
```
All of the above are valid and I really don't want anyone to remember whether `order` comes before `by` or the other way around.
Compare to zip – the `function` argument is defaulted to `[_, _]` (i.e. just do the normal zip, by shoving both elements into a pair):
```
[1,2,3].zip [4,5,6] # ==> [[1,4], [2,5], [3,6]]
[1,2,3].zip [4,5,6] (+) # ==> [5, 7, 9], the meaning of this expression is very clear – `zip` only takes one function.
[1,2,3].zip [4,5,6] function=(+) # ==> same result, but it looks very awkward to me
```

Reply:
```suggestion
    # Methods with default parameters can be called with or without the parameter names.
    # When calling a method with named parameters, the order of the parameters does not matter.
```

Reply:
I think it may be worth thinking about the orthogonality of named arguments vs. defaulted arguments.
The truth is:
1. Any argument may be called by position or by name
2. Any argument may be defaulted or not
The proposed change seems to tie named arguments to defaulted arguments (i.e. suggests that only defaulted arguments can be called by name)

Reply:
OH, that's interesting, so if I did say
`sum x y = x + y`
I could do 
`z = sum x=1 y=2` ?

It ties named arguments to defaulted arguments because it never occurred to me that might not be the case :joy:

Reply:
correct, the defaultedness has nothing to do with named arguments, other than it so happens that the arguments we call by name are most often defaulted

Reply:
```suggestion
    # Defaulted parameters can be omitted when calling methods.
```